git_machete/cmd.py

```diff
@@ -1570,15 +1573,174 @@ def flush_caches(self) -> None:
+            raise MacheteException(f'No PR is opened in `{org}/{repo}` for branch `{head}`')
+        debug(f'retarget_github_pr({head})', f'found {pr}')
+
+        new_base: Optional[str] = self.up_branch.get(head)
+        if not new_base:
+            raise MacheteException(f'Branch `{head}` does not have a parent branch (it is a root) '
+                                   f'even though there is an open PR #{pr.number} to `{pr.base}`.\n'
+                                   f'Consider modifying the branch definition file (`git machete edit`) so that `{head}` is a child of `{pr.base}`.')
+
+        if pr.base != new_base:
+            set_base_of_pull_request(org, repo, pr.number, base=new_base)
+            print(fmt(f'The base branch of PR #{pr.number} is switched to `{new_base}`'))
+        else:
+            print(fmt(f'The base branch of PR #{pr.number} is already `{new_base}`'))
+
+    def __derive_remote_and_github_org_and_repo(self) -> Tuple[str, Tuple[str, str]]:
+        url_for_remote: Dict[str, str] = {
+            remote: self.__git.get_url_of_remote(remote) for remote in self.__git.remotes()
+        }
+        if not url_for_remote:
+            raise MacheteException(fmt('No remotes defined for this repository (see `git remote`)'))
+
+        org_and_repo_for_github_remote: Dict[str, Tuple[str, str]] = {
+            remote: parse_github_remote_url(url) for remote, url in url_for_remote.items() if is_github_remote_url(url)
+        }
+        if not org_and_repo_for_github_remote:
+            raise MacheteException(
+                fmt('Remotes are defined for this repository, but none of them corresponds to GitHub (see `git remote -v` for details)'))
+
+        if len(org_and_repo_for_github_remote) == 1:
+            return list(org_and_repo_for_github_remote.items())[0]
+
+        if 'origin' in org_and_repo_for_github_remote:
+            return 'origin', org_and_repo_for_github_remote['origin']
+
+        raise MacheteException(f'Multiple non-origin remotes correspond to GitHub in this repository: '
+                               f'{", ".join(org_and_repo_for_github_remote.keys())}, aborting')
+
+    def create_github_pr(self, head: str, draft: bool) -> None:
+        # first make sure that head branch is synced with remote
+        self.__sync_before_creating_pr()
+
+        base: Optional[str] = self.up_branch.get(head)
+        if not base:
+            raise MacheteException(
+                f'Branch {head} does not have a parent branch (it is a root), base branch for the PR cannot be established')
+
+        org: str
+        repo: str
+        _, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        current_user: Optional[str] = derive_current_user_login()
+        debug(f'create_github_pr({head})', f'organization is {org}, repository is {repo}')
+        debug(f'create_github_pr({head})', 'current GitHub user is ' + (current_user or '<none>'))
+
+        def slurp_file_or_empty(path: str) -> str:
```

Comment:
Maybe move to `utils` + also I think `read()` call should be wrapped into `with` (so that `close()` is always called, even in case of an error)